src/components/pages/dashboard-pages/Notifications.jsx

```diff
@@ -1,10 +1,194 @@
+    };
+
+    getStudentInfo();
+  }, [classes, userData]);
+
+  const sortClassesByApplicationDate = (classes) => {
+    return classes.map(classInfo => ({
+      ...classInfo,
+      applications: classInfo.applications.slice().sort((a, b) => new Date(b.appliedAt) - new Date(a.appliedAt)) // Descending
+    }));
+  };
+
+  const sortClassesByEarliestApplicationDate = (classes) => {
+    return classes.slice().sort((a, b) => {
+      const earliestDateA = a.applications.length > 0 ? new Date(Math.min(...a.applications.map(app => new Date(app.appliedAt)))) : new Date();
+      const earliestDateB = b.applications.length > 0 ? new Date(Math.min(...b.applications.map(app => new Date(app.appliedAt)))) : new Date();
+
+      return earliestDateB - earliestDateA; // Descending order
+    });
+  };
+
+  const formatDateWithWeekday = (dateString) => {
+    const date = new Date(dateString);
+    const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+    return date.toLocaleDateString('en-US', options);
+  };
+
+  const formatDateWithoutWeekday = (dateString) => {
+    const date = new Date(dateString);
+    const options = { year: 'numeric', month: 'long', day: 'numeric' };
+    return date.toLocaleDateString('en-US', options);
+  };
+
+  const calculateAge = (dateOfBirth) => {
+    const birthDate = new Date(dateOfBirth);
+    const today = new Date();
+    let age = today.getFullYear() - birthDate.getFullYear();
+    const monthDifference = today.getMonth() - birthDate.getMonth();
+
+    if (monthDifference < 0 || (monthDifference === 0 && today.getDate() < birthDate.getDate())) {
+      age--;
+    }
+
+    return age;
+  };
+
+  const handleApprove = (classId, applicationId) => {
+    console.log('Approved:', classId, applicationId);
+  };
+
+  const handleDecline = (classId, applicationId) => {
+    console.log('Declined:', classId, applicationId);
+  };
+
+  if (userData.role !== 'teacher') {
+    return <p>Coming soon...</p>;
+  }
+
   return (
-    <>
+    <div className="container mt-10">
       <h1 className="text-red font-bold text-2xl sm:text-4xl font-spartan uppercase">
-        Notifications page - common
+        New applications:
       </h1>
-    </>
+      {classesError && <p className="text-red-500">{classesError.message}</p>}
+      <div className="flex flex-col gap-6 mt-10 m-5 md:m-10">
+        {classes.length > 0 ? (
+          classes.map((classInfo) => (
+            <div
+              key={classInfo._id}
+              className="border rounded-lg border-gray p-5 bg-pureWhite"
+            >
+              <h2 className="font-roboto font-medium text-2xl mb-4">
+                {classInfo.classTitle}
+              </h2>
+              <hr className="my-4 border-gray" />
+              {classInfo.applications.map((application) => {
+                const applicantDetails = applicants.find(
+                  (user) => user._id === application.userId
+                );
+                return (
```

Comment:
This section of your JSX represents the application section. This could be its own component. And those functions it relies on can be easily imported from their respective utils folder.

Reply:
I will refactor this part of the code into a separate PR.

Reply:
Great idea!